From 62ae069bc8c559f4814f64504335475edc522622 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavel=20B=C5=99ezina?= <pbrezina@redhat.com>
Date: Tue, 19 Jan 2016 13:30:58 +0100
Subject: [PATCH 45/55] sdap_connect_send: fail if uri or sockaddr is NULL
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Resolves:
https://fedorahosted.org/sssd/ticket/2904

Reviewed-by: Michal Židek <mzidek@redhat.com>
(cherry picked from commit 8bd9ec3a8885b01a34863d22aa784e221fc422fb)
---
 src/providers/ldap/sdap_async_connection.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/providers/ldap/sdap_async_connection.c b/src/providers/ldap/sdap_async_connection.c
index 8f5227d263f995693f6e65bd238171538aa52af7..85b7aaa5bf5acedf3511ffe6f8636be007d5a136 100644
--- a/src/providers/ldap/sdap_async_connection.c
+++ b/src/providers/ldap/sdap_async_connection.c
@@ -75,6 +75,12 @@ struct tevent_req *sdap_connect_send(TALLOC_CTX *memctx,
     req = tevent_req_create(memctx, &state, struct sdap_connect_state);
     if (!req) return NULL;
 
+    if (uri == NULL || sockaddr == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "Invalid uri or sockaddr\n");
+        ret = EINVAL;
+        goto fail;
+    }
+
     state->reply = talloc(state, struct sdap_msg);
     if (!state->reply) {
         talloc_zfree(req);
-- 
2.5.0

